### DISCUSSION TITLE
Support of new MB0 series with 2 Luna batteries

### DISCUSSION THREAD

**Discussion by philkunz** [wlcrs/huawei_solar#669] [Connecting to the inverter]
### Describe your feature request

Hi,

thank you for the great work on this integration.
I have this working with an MB1 setup.

Do you happen to know (or a feeling) if the new MB0 series with 2 Luna Batteries works with this integration?
https://solar.huawei.com/en/professionals/all-products/SUN2000-12-15-17-20-25K-MB0/specs

Thank you.

### Proper usage

- [X] I confirm that this is not a bug report or support request
- [X] I confirm that this feature request is within the stated scope of the integration

**Comment by Roving-Ronin** [comment]
@philkunz 

Compare the Modbus Register info for these inverters ( https://support.huawei.com/enterprise/en/doc/EDOC1100351583?section=k001 ) against the Modbus Registers v05 that is the latest version that all the other inverters also use. From a quick glance there are a number of registers that are new for the new MB0 / ZHMBO / M5 inverters that use the same firmware, but these seem to be registers to query things like what software version / apps etc its running. 
From a casual check it looks like the registers that are used by integration to query the devices, are the same registers between these new inverters and the current L1 / M1 / M0 / M2 / M3 etc inverters.

**Comment by tmenguy** [comment]
For me I have a brand new MBO , and the integration works pretty well, I have no battery for now so can't comment on this part. (Connection via ethernet Dongle is working well)

**Comment by philkunz** [comment]
Thank you so much guys. Those comments help a lot. Looks like it is a go.
I'll report back once the system is installed.

**Comment by tmenguy** [comment]
I was surprised but the port was 502 on the Dongle (and not 6607) with latest firmware it seems

**Comment by Roving-Ronin** [comment]
@tmenguy @philkunz 

Hi,

Could you gents please confirm what your configuration is, and confirm that all is operating correctly, so that it might be posted on the Confirmed Firmware wiki page? i.e.

Inverter Type:   SUN2000-??KTL-MB0
Inverter Firmware version: V200R023C10SPC201 (if your using the latest release)

SDongle present: Yes / No
SDongle Firmware Version: 

Power meter present: Three Phase

Battery: LG RESU xxkWh / LUNA2000 xxkWh / None
Battery Firmware version:  V100R002C00SPC127 (if your using a LUNA and on the latest release)
Number of Battery(s): 

Connection Method: Connecting to WiFi or Ethernet on the sDongle over LAN. Or connecting to inverter inbuilt WiFi AP, or via Elfin modbus relay etc?

**Comment by tmenguy** [comment]
Ok so for me everything working well on 1.40a2 , Home Assitant 2024.4.3

inverter : SUN2000-15K-MBO, firmware: V200R023C10SPC200

sDongle : yes,  SDongleA-05  Firmware V100R001COQSPC125

no battery

connection : SDongle through wired ethernet (very solid connection)

**Reply by Roving-Ronin** [reply]
@tmenguy 
Thanks, updated into the Wiki Confirmed Firmware.
BTW would you mind testing some code for me? In the Developer Tools --> Template, could you please paste this:
```
{{ device_attr("sensor.inverter_active_power", "model") }}
```
and let me know what response is returned (right side screen)?  Just wanting to confirm the model number / format that is being returned for the MB0 models. Thanks.

**Reply by tmenguy** [reply]
Result:

SUN2000-15K-MB0

**Reply by Roving-Ronin** [reply]
Thanks for that...  looks like the buggers have changed the naming format to drop the 'TL' from what used to be 'KTL'. Back to fix up my regex to detect the inverter models.

**Reply by tmenguy** [reply]
Ah ah yes ... those software guys :)
